model-server/src/test/kotlin/org/modelix/model/server/handlers/ContentExplorerTest.kt

```diff
@@ -71,4 +77,31 @@ class ContentExplorerTest {
         val response = client.get("/content/$versionHash/$nodeId/")
         assertTrue(response.successful)
     }
+
+    @Test
+    fun `nodes can be expanded`() = runTest {
+        val client = createClient {
+            install(ClientContentNegotiation) { json() }
+        }
+
+        val delta: VersionDelta = client.post("/v2/repositories/node-expansion/init").body()
+
+        val versionHash = delta.versionHash
+        val version = CLVersion.loadFromHash(versionHash, modelClient.storeCache)
+        val nodeId = checkNotNull(version.getTree().root?.id)
+
+        val expandedNodes = ContentExplorerExpandedNodes(setOf(nodeId.toString()), false)
+
+        val response = client.post("/content/$versionHash/") {
+            contentType(ContentType.Application.Json)
+            setBody(expandedNodes)
+        }
+
+        val html = Input.fromString(response.bodyAsText().replace("<br>", "")).build() // br leads to parser exception
```

Comment:
The BR thing probably happens because you use a strict XML parse and not a parser specifically tailored to HTML The next change to the HTML structure can result in the next error of this kind. It would be better to switch to dedicated HTML parsing to avoid these issues: https://stackoverflow.com/a/28307130/283649